src/mlpack/methods/reinforcement_learning/replay/random_replay.hpp

```diff
@@ -105,6 +140,37 @@ class RandomReplay
     }
   }
 
+  /**
+   * Get the reward, next state and terminal boolean for nth step.
+   *
+   * @param reward Given reward.
+   * @param nextState Given next state.
+   * @param isEnd Whether next state is terminal state.
+   * @param discount The discount parameter.
+   */
+  void GetNStepInfo(double& reward,
+                    StateType& nextState,
+                    bool& isEnd,
+                    const double& discount)
+  {
+    reward = nStepBuffer.back().reward;
+    nextState = nStepBuffer.back().nextState;
+    isEnd = nStepBuffer.back().isEnd;
+
+    // Should start from the second last transition in buffer.
+    for (int i = nStepBuffer.size() - 2; i >= 0; i--)
+    {
+      double r = nStepBuffer[i].reward;
+      bool iE = nStepBuffer[i].isEnd;
+      reward = r + discount * reward * (1 - iE);
```

Comment:
Maybe we can directly use `nStepBuffer[i].reward` here since it's one used once?

Reply:
yeah, thats right!